surfactant/plugin/manager.py

```diff
@@ -65,6 +66,25 @@ def _register_plugins(pm: pluggy.PluginManager) -> None:
     for plugin in internal_plugins:
         pm.register(plugin)
 
+    config_manager = ConfigManager()
+
+    # Retrieve the current list of blocked plugins
+    current_blocked_plugins = config_manager.get("core", "disable_plugins", [])
+    for plugin_name in current_blocked_plugins:
+        # Check if the plugin is already blocked
+        if pm.is_blocked(plugin_name):
+            print(f"Plugin '{plugin_name}' is already disabled.")
+            continue
+
+        # Unregister the plugin
+        plugin = pm.unregister(name=plugin_name)
+        if plugin is None:
+            print(f"Plugin '{plugin_name}' not found.")
```

Comment:
```suggestion
            logger.info(f"Disabled plugin '{plugin_name}' not found.")
```

Log using logger instead of print. Actually not sure if this is a message that a user necessarily cares about either, other than for the purposes of cleaning up their disabled plugin list.